### ISSUE TITLE
Fatal error accessing workflow as copyeditor

### ISSUE THREAD

**Issue by NateWr** [pkp/pkp-lib#7822]
**Describe the bug**
When assigned as a copyeditor, I can not access the submission workflow because of a fatal error. This only effects PHP 8.0 or above because `array_intersect` is more strict about the arguments it accepts.

**To Reproduce**
1. Run a site in PHP 8.0 or above.
2. Assign a copyeditor to a submission.
3. Login as the copyeditor.
4. Try to view the submission workflow.
5. See error

**What application are you using?**
Effects `main` and `stable-3_3_0`. This fix should be backported to 3.3.0, because eventually it will need to support PHP 8.0.

**Additional information**
The error:

```
PHP Fatal error:  Uncaught TypeError: array_intersect(): Argument #2 must be of type array, null given in lib/pkp/pages/workflow/PKPWorkflowHandler.inc.php:167
Stack trace:
#0 lib/pkp/pages/workflow/PKPWorkflowHandler.inc.php(167): array_intersect(Array, NULL)
#1 [internal function]: PKPWorkflowHandler->index(Array, Object(APP\core\Request))
#2 lib/pkp/classes/core/PKPRouter.inc.php(465): call_user_func(Array, Array, Object(APP\core\Request))
#3 lib/pkp/classes/core/PKPPageRouter.inc.php(308): PKP\core\PKPRouter->_authorizeInitializeAndCallRequest(Array, Object(APP\core\Request), Array, false)
#4 lib/pkp/classes/core/Dispatcher.inc.php(163): PKP\core\PKPPageRouter->route(Object(APP\core\Request))
#5 lib/pkp/classes/core/PKPApplication.inc.php(376): PKP\core\Dispatcher->dispatch(Object(APP\core\Request))
#6 index.php(70): PKP\core\PKPApplication->execute()
#7 {main}
  thrown in lib/pkp/pages/workflow/PKPWorkflowHandler.inc.php on line 167
```

**Comment by asmecher** [comment]
@NateWr, FYI, I think this is a PHP8.1 and up issue. I've been tracking/fixing these in https://github.com/pkp/pkp-lib/issues/7690.

**Comment by NateWr** [comment]
I've run into it on PHP 8.0.17.

**Comment by asmecher** [comment]
Yes, but I didn't see the problem on my machine (in fact, not using PHP8.1.2 either). Are you using one of the test dataset accounts? I'm using mfritz.

**Comment by mariuszsienkiewicz** [comment]
So is it only a need to change this:
```php
$canAccessProduction = (bool) array_intersect($editorialWorkflowRoles, $accessibleWorkflowStages[WORKFLOW_STAGE_ID_PRODUCTION]);
``` 

into this:

```php
$canAccessProduction = (bool) array_intersect($editorialWorkflowRoles, $accessibleWorkflowStages[WORKFLOW_STAGE_ID_PRODUCTION]  ?? []);
```

in a couple of places?

**Comment by NateWr** [comment]
Yes, I think so.

**Comment by asmecher** [comment]
Thanks, @NateWr and @mariuszsienkiewicz -- I've transformed your suggestion into a couple of pull requests, Mariusz:
- pkp-lib `main` branch: https://github.com/pkp/pkp-lib/pull/7861 (for release in 3.4.0)
- pkp-lib `stable-3_3_0` branch: https://github.com/pkp/pkp-lib/pull/7862 (to apply against 3.3.0-x older than 3.3.0-11)

The appropriate PR/patch should be applied in the `lib/pkp` subdirectory.

If either of you can confirm that this resolves the issue, I can commit it!

**Comment by NateWr** [comment]
I can confirm this resolved the issue in `stable-3_3_0`.

**Comment by asmecher** [comment]
This was actually fixed in 3.3.0-11 -- reopening that milestone temporarily to correct the record, and closing this issue. Thanks all!
